MyString.java

```diff
@@ -36,7 +56,29 @@ public static int countChar(String str, char ch) {
      * @return true is str1 is a subset of str2, false otherwise
      */
     public static boolean subsetOf(String str1, String str2) {
-         //// Replace the following statement with your code
+        if (str1 == "") {
+            return true;
+        }
+        int counter = 0;
+        String newStr1 = "";
+        
+
+        for(int i=0; i < str1.length(); i++){
+            char charOfStr1 = (char) str1.charAt(i);
+            if (countChar(newStr1, charOfStr1) == 0) {
+                newStr1 = newStr1 + charOfStr1;
+            }
+        }
+
+        for(int i=0; i < newStr1.length(); i++){
+            char charOfStr1 = (char) newStr1.charAt(i);
+            if (countChar(str2, charOfStr1) == 1) {
+                counter++;
+            }
+            if (counter == str1.length()) {
+                return true;
+            }
+        }        

```

Comment:
I see what you did here, 
It could be structured a bit better
but I idea is very good.. 

consider the following
checking first appearance of char in string is in index could be written as follows
`if ( i == str.indexOf(str.charAt(i)))`